woocommerce/woocommerce-free-shipping-progress-bar-block
# Implement @woocommerce/price-format dependency

Fixes #28

### Note

The remaining price should be formatted using the @woocommerce/price-format dependency, which is included in the WooCommerce Blocks plugin.

### Testing instructions

#### Before all test cases

1. Check out this PR.
2. Create a test page and add the Cart block.
3. Insert the Free Shipping Progress Bar block into the Cart block and save the page.
4. Set the free shipping value within the  Free Shipping Progress Bar block to 1200.
5. Head over to the frontend and add products to your cart.
6. Ensure the cart total is higher than 1000 (to test the thousand separator) and below 1200 (to see the remaining price).

#### Test 0 decimals currency

1. Go to `WP Admin » WooCommerce » Settings » Currency options` and use the following settings:
    * Currency: `Indonesian rupiah (Rp)`
    * Currency position: `Left`
    * Thousand separator: `.`
    * Decimal separator: `,`
    * Number of decimals: `0`
2. Go to the cart and verify that the price is formatted as expected.

<img width="500" alt="" src="https://user-images.githubusercontent.com/3323310/143674092-fbb608e8-0d9a-48dd-996d-ef72d43cec56.png">

#### Test 2 decimals currency

1. Go to `WP Admin » WooCommerce » Settings » Currency options` and use the following settings:
    * Currency: `USD $`
    * Currency position: `Left + space`
    * Thousand separator: `,`
    * Decimal separator: `.`
    * Number of decimals: `2`
2. Go to the cart and verify that the price is formatted as expected.

<img width="500" alt="" src="https://user-images.githubusercontent.com/3323310/143674102-5b488187-fb52-4bd2-9a46-d387be49c5d3.png">

#### Test 3 decimals currency

1. Go to `WP Admin » WooCommerce » Settings » Currency options` and use the following settings:
    * Currency: `Libyan Dinar`
    * Currency position: `Right + space`
    * Thousand separator: `.`
    * Decimal separator: `,`
    * Number of decimals: `3`
2. Go to the cart and verify that the price is formatted as expected.

<img width="500" alt="" src="https://user-images.githubusercontent.com/3323310/143674099-03522d4e-f09f-47ba-b1be-a7c8484be613.png">
